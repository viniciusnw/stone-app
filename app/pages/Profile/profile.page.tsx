import React, { Component } from "react";
import { View, Text, ActivityIndicator } from "react-native";

import { Button, Icon, Logo } from "@Components";
import { useNavigation } from "@react-navigation/native";

export default function Profile(props: any) {
  const navigation = useNavigation();

  React.useEffect(
    () =>
      props.setState({
        bottomBar: {
          menu: [
            {
              icon: "wallet",
              label: "Carteira",
              onPress: () => true,
              primary: false,
            },
            {
              icon: "home",
              label: "Home",
              onPress: () => props.goTo("Home"),
              primary: true,
            },
            {
              icon: "logout",
              label: "Logout",
              onPress: () => true,
              primary: false,
            },
          ],
        },
        topBar: {
          title: "Profile",
          back: navigation.goBack,
        },
      }),
    []
  );

  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Text />
    </View>
  );
}
