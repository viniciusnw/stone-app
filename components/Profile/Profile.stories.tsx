import React from 'react';
import { Profile } from '@Components';

const user = {
  src: 'https://www.graciemag.com/wp-content/uploads/2012/06/house.jpg',
  name: 'Bruce Wayne',
  rule: 'Wayne Interprise',
};

export const Default = {};

export default {
  title: "Profile",
  component: () => (
    <Profile {...user} />
  ),
};
