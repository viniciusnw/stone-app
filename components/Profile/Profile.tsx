import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components/native';
import { View, Text, Image } from 'react-native';

const StyledContainer = styled(View)`
  width: 100%;
  align-items: center;
  border: 1px solid transparent;
  border-bottom-color: ${({ theme }) => theme.color.border};
  padding: 42px 0;
`;

const Picture = styled(Image)`
  width: 90px;
  height: 90px;
  border-radius: 100px;
  margin-bottom: 20px;
`;

const StyledName = styled(Text)`
  font-size: 20px;
  font-weight: 700;
  color: ${({ theme }) => theme.color.black};
`;

const StyledRule = styled(Text)`
  margin-top: 5px;
  font-weight: 700;
  font-size: 14px;
  color: ${({ theme }) => theme.color.gray};
`;

type Props = {
  src: string,
  name: string,
  rule: string,
}

const Profile: React.FC<Props>  = ({ src, name, rule, ...props }: any) => {
  return (
    <StyledContainer {...props}>
      <Picture source={{ uri: src }} />
      <StyledName>{name}</StyledName>
      <StyledRule>{rule}</StyledRule>
    </StyledContainer>
  );
};


export default Profile;
