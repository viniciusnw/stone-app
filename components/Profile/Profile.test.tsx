import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Profile from "./Profile";

jest.useFakeTimers();

const user = {
  src: "https://i.pinimg.com/280x280_RS/39/fd/09/39fd09a3f0962e5b7b4a3ccbf5a61057.jpg",
  name: "Bruce Wayne",
  rule: "Wayne Interprise",
};

test("Profile", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Profile {...user} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
