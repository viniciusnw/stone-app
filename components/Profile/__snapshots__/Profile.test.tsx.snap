// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#ddd",
      "borderColor": "transparent",
      "borderStyle": "solid",
      "borderWidth": 1,
      "paddingBottom": 42,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 42,
      "width": "100%",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://i.pinimg.com/280x280_RS/39/fd/09/39fd09a3f0962e5b7b4a3ccbf5a61057.jpg",
      }
    }
    style={
      {
        "borderBottomLeftRadius": 100,
        "borderBottomRightRadius": 100,
        "borderTopLeftRadius": 100,
        "borderTopRightRadius": 100,
        "height": 90,
        "marginBottom": 20,
        "width": 90,
      }
    }
  />
  <Text
    style={
      {
        "color": "#111111",
        "fontSize": 20,
        "fontWeight": "700",
      }
    }
  >
    Bruce Wayne
  </Text>
  <Text
    style={
      {
        "color": "#666666",
        "fontSize": 14,
        "fontWeight": "700",
        "marginTop": 5,
      }
    }
  >
    Wayne Interprise
  </Text>
</View>
`;
